### DISCUSSION TITLE
The on_row_dblclick event of table cannot be triggered

### DISCUSSION THREAD

**Discussion by hu76589** [zauberzeug/nicegui#1718] [Q&A] [answered]
### Question

there is a table in the page,i want to double-click on the row, and then do something. 
This is an example
```py
from nicegui import globals, ui

globals.get_client().content.classes('absolute-full')

def on_row_dblclick(evt, row, index):
    ui.notify('on_row_dblclick!')

with ui.column().classes('w-full h-full'):
    columns = [
        {'name': 'name', 'label': 'Name', 'field': 'name', 'required': True, 'align': 'left'},
        {'name': 'age', 'label': 'Age', 'field': 'age'},
    ]
    rows = [
        {'name': 'Alice', 'age': 18},
        {'name': 'Bob', 'age': 21},
        {'name': 'Carol', 'age': 21},
    ]
    ui.table(columns=columns, rows=rows, row_key='name') \
        .classes('w-full h-full') \
        .on('row-dblclick', on_row_dblclick)

ui.run()
```
When I double-click on a row, on_row_dblclick(evt, row, index) Not running.
What do I need to do？
thanks!

**Comment by falkoschindler** [comment, marked as answer]
Hi @hu76589,
The event name needs to be written in camel case like this:
```py
.on('rowDblclick', on_row_dblclick)
```

**Reply by hu76589** [reply]
thanks!

and i want to get the row or row_index like this
```
evt.args[-2] # get row
evt.args[-1] # get row index
```
Is there a better way？

**Reply by falkoschindler** [reply]
I'd use positive indices:
```py
row = evt.args[1]
index = evt.args[2]
```
But yes, that's how to access generic event arguments.

**Comment by retsyo** [comment]
sorry, 2 questions
- what is the proper way to write `.on('rowDblclick', on_row_dblclick)`? now the above code says `TypeError: on_row_dblclick() missing 2 required positional arguments: 'row' and 'index'`
- how to get cells' value in the double-clicked row?

thanks

**Reply by falkoschindler** [reply]
Here is an updated example for a more recent NiceGUI version:
```py
def on_row_dblclick(e: events.GenericEventArguments):
    _, row, index = e.args
    print(f'{row=}')
    print(f'{index=}')


columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name'},
    {'name': 'age', 'label': 'Age', 'field': 'age'},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol', 'age': 21},
]
ui.table(columns=columns, rows=rows, row_key='name') \
    .on('row-dblclick', on_row_dblclick)
```
Note that we only get the row, not the individual cell that has been clicked on.

**Reply by retsyo** [reply]
Thanks

A summary, on https://quasar.dev/vue-components/table, we can find `Events` in `QTable API` as 
```javascript
@row-dblclick
: (evt, row, index) => void
```

So 
- we change `row-dblclick` into camel case `rowDblclick` which should be used in `tab.on('rowDblclick', callbackFunction)` 
- `e.args` gives `(evt, row, index)`

**Reply by falkoschindler** [reply]
@retsyo Since version 1.4.6 you can also use kebab-case: "row-dblclick".

**Reply by retsyo** [reply]
Aha, I am using 1.4.5 since I am too lazy to clean my code to put `add_body_html` before the page has been sent to the client
